mdn/src/test/java/org/apache/james/mdn/MDNFactoryTest.java

```diff
@@ -537,4 +538,66 @@ public void generateMDNReportThrowOnNullDisposition() {
             .originalRecipientField(new OriginalRecipient("originalRecipient"))
             .build();
     }
+
+    @Test
+    public void generateMDNReportShouldFormatGateway() {
+        Disposition disposition = Disposition.builder()
+            .actionMode(DispositionActionMode.Automatic)
+            .sendingMode(DispositionSendingMode.Automatic)
+            .type(DispositionType.Processed)
+            .addModifier(DispositionModifier.Error)
+            .addModifier(DispositionModifier.Failed)
+            .build();
+
+        String report = MDNReport.builder()
+            .reportingUserAgentField(new ReportingUserAgent(
+                "UA_name",
+                Optional.of("UA_product")))
+            .gatewayField(new Gateway("host.com"))
+            .finalRecipientField(new FinalRecipient(Optional.of("final_recipient")))
+            .originalRecipientField(new OriginalRecipient("originalRecipient"))
+            .originalMessageIdField(new OriginalMessageId(Optional.of("original_message_id")))
+            .dispositionField(disposition)
+            .build()
+            .formattedValue();
+
+        assertThat(report)
+            .isEqualTo("Reporting-UA: UA_name; UA_product\r\n" +
```

Comment:
usage of isEqualTo doesn't make sense : you want to make sure Gateway is present with a given content and maybe where it's located but not more than that.

Reply:
Here I value the fact of having explicit full MDN values, to show real word example with real world value.

If I am really asked for it, I will reduce it to a simple **contains** call, but I would prefer not to.

Reply:
I attempted to start a discussion on this in your earlier comments.